src/results/result-form-field.component.tsx

```diff
@@ -29,20 +32,39 @@ const ResultFormField: React.FC<ResultFormFieldProps> = ({ concept, control }) =
 
   return (
     <>
-      {isTextOrNumeric(concept) && (
+      {isText(concept) && (
         <Controller
           control={control}
           name={concept.uuid}
           render={({ field }) => (
             <TextInput
-              key={concept.uuid}
               className={styles.textInput}
+              id={concept.uuid}
+              key={concept.uuid}
+              labelText={concept?.display ?? ''}
+              type="text"
               {...field}
-              type={concept.datatype.display === 'Numeric' ? 'number' : 'text'}
-              labelText={
-                concept?.display + (concept.datatype.display === 'Numeric' ? printValueRange(concept) ?? '' : '')
-              }
-              autoFocus
+            />
+          )}
+        />
+      )}
+
+      {isNumeric(concept) && (
+        <Controller
+          control={control}
+          name={concept.uuid}
+          render={({ field }) => (
+            <NumberInput
+              allowEmpty
+              className={styles.numberInput}
+              disableWheel
+              hideSteppers
+              id={concept.uuid}
+              key={concept.uuid}
+              label={concept?.display + printValueRange(concept)}
+              onChange={(event) => field.onChange(event.target.value)}
+              step={0.01}
```

Comment:
@ibacher @ojwanganto is a step precision value of 0.01 for these number inputs reasonable?

Reply:
This doesn't look appropriate for all use cases. If we can compute this based on the concept in question then we can have much control. For instance, a test that accepts decimals should infer a lot from the configured min and max values but we can hear @ibacher's thoughts